mergify_engine/tests/functional/base.py

```diff
@@ -622,6 +625,18 @@ async def create_pr(
             await self.git("commit", "--no-edit", "-m", f"{title}, moved")
         await self.git("push", "--quiet", base_repo, branch)
 
+        if verified:
+            temporary_folder = tempfile.mkdtemp()
+            os.environ["GNUPGHOME"] = temporary_folder
+            subprocess.run(["gpg", "--import"], input=config.TESTING_GPGKEY_SECRET)
+            await self.git("config", "user.signingkey", config.TESTING_ID_GPGKEY_SECRET)
+            await self.git(
+                "config", "user.email", "engineering+mergify-test@mergify.io"
+            )
+            await self.git(
```

Comment:
Instead of amending the commit on top of the branch and changing the message, we should just add "-S" to previous git `commit command`.

Reply:
You mean the first commit made ?

Reply:
Even all commits made by `create_pr()`.